cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -1316,12 +1335,30 @@ private class ProposalExecutionRunnable implements Runnable {
       _replicationThrottle = replicationThrottle;
       _isTriggeredByUserRequest = isTriggeredByUserRequest;
       _lastSlowTaskReportingTimeMs = -1L;
+      if (removedBrokers != null && !removedBrokers.isEmpty()) {
+        _executionTimerInvolveBrokerRemovalOrDemotion = _proposalExecutionTimerInvolveBrokerRemoval;
+      } else if (demotedBrokers != null && !demotedBrokers.isEmpty()) {
+        _executionTimerInvolveBrokerRemovalOrDemotion = _proposalExecutionTimerInvolveBrokerDemotionOnly;
+      } else {
+        _executionTimerInvolveBrokerRemovalOrDemotion = null;
+      }
     }
 
     public void run() {
       LOG.info("Starting executing balancing proposals.");
-      UserTaskManager.UserTaskInfo userTaskInfo = initExecution();
-      execute(userTaskInfo);
+      final long start = System.nanoTime();
```

Comment:
You probably don't want the nanoseconds as the unit. We usually use milliseconds.

Change it to something like` final long startTimeMs = System.miliseconds();`

Reply:
done